gcloud-java-compute/src/test/java/com/google/gcloud/compute/AddressTest.java

```diff
@@ -0,0 +1,284 @@
+    assertSame(serviceMockReturnsOptions, address.compute());
+    assertNull(address.address());
+    assertNull(address.creationTimestamp());
+    assertNull(address.description());
+    assertNull(address.id());
+    assertNull(address.status());
+    assertNull(address.usage());
+    address = new Address.Builder(serviceMockReturnsOptions, REGION_ADDRESS_ID).build();
+    assertEquals(REGION_ADDRESS_ID, address.addressId());
+    assertSame(serviceMockReturnsOptions, address.compute());
+    assertNull(address.address());
+    assertNull(address.creationTimestamp());
+    assertNull(address.description());
+    assertNull(address.id());
+    assertNull(address.status());
+    assertNull(address.usage());
+  }
+
+  @Test
+  public void testToBuilder() {
+    initializeExpectedAddress(16);
+    compareAddress(instanceAddress, instanceAddress.toBuilder().build());
+    compareAddress(globalForwardingAddress, globalForwardingAddress.toBuilder().build());
+    compareAddress(regionForwardingAddress, regionForwardingAddress.toBuilder().build());
+    Address newAddress = instanceAddress.toBuilder().description("newDescription").build();
+    assertEquals("newDescription", newAddress.description());
+    newAddress = newAddress.toBuilder().description("description").build();
+    compareAddress(instanceAddress, newAddress);
+  }
+
+  @Test
+  public void testToAndFromPb() {
+    initializeExpectedAddress(20);
+    compareAddress(globalForwardingAddress,
+        Address.fromPb(serviceMockReturnsOptions, globalForwardingAddress.toPb()));
+    compareAddress(regionForwardingAddress,
+        Address.fromPb(serviceMockReturnsOptions, regionForwardingAddress.toPb()));
+    compareAddress(instanceAddress,
+        Address.fromPb(serviceMockReturnsOptions, instanceAddress.toPb()));
+    Address address = new Address.Builder(serviceMockReturnsOptions, GLOBAL_ADDRESS_ID).build();
+    compareAddress(address, Address.fromPb(serviceMockReturnsOptions, address.toPb()));
+    address = new Address.Builder(serviceMockReturnsOptions, REGION_ADDRESS_ID).build();
+    compareAddress(address, Address.fromPb(serviceMockReturnsOptions, address.toPb()));
+  }
+
+  @Test
+  public void testDeleteOperation() {
+    initializeExpectedAddress(4);
+    expect(compute.options()).andReturn(mockOptions);
+    Operation operation = new Operation.Builder(serviceMockReturnsOptions)
+        .operationId(GlobalOperationId.of("project", "op"))
+        .build();
+    expect(compute.delete(REGION_ADDRESS_ID)).andReturn(operation);
+    replay(compute);
+    initializeAddress();
+    assertSame(operation, address.delete());
+  }
+
+  @Test
+  public void testDeleteNull() {
+    initializeExpectedAddress(3);
+    expect(compute.options()).andReturn(mockOptions);
+    expect(compute.delete(REGION_ADDRESS_ID)).andReturn(null);
+    replay(compute);
+    initializeAddress();
+    assertNull(address.delete());
+  }
+
+  @Test
+  public void testExists_True() throws Exception {
```

Comment:
There's no real widely recognized convention on test names that I am aware of. Amongst the most popular I've seen are:

1. `test[NameOfTheFunctionalityInCamelCase]`
2. `[method]_[condition]_[returnValue]`

What I used here is none of the two but I prefer it over `testExistsAndReturnTrue` (which should be the name according to 1.) and `exists_true` (which should be the name according to 2.) The latter also breaks the naming we used so far.

My point was: I agree that you might not like it and we might want to change but since this is used also elsewhere we should open a global issue and discuss it there before changing it here. This should be a common practice to avoid missing issues that apply to code already merged.